editor/components/block-list/block.js

```diff
@@ -628,6 +606,28 @@ export class BlockListBlock extends Component {
 					aria-label={ blockLabel }
 					data-block={ block.uid }
 				>
+					<BlockDropZone
```

Comment:
I know this is not pretty, and I will need some help making sure this works as well as it needs to. But the reason for moving all of this is that it's necessary in order to allow the margins to collapse. Quite simply, all these elements have to be children of the _content_ element itself, otherwise the content elements margins will be uncollapsed by the presence of adjecent divs. See also this codepen: https://codepen.io/joen/pen/aqKLqa?editors=1100

Reply:
I don't know exactly why these components are outside of this div but what I know is that this block's content wrapper is targeted in code in a lot of places, so this change could have side effects. I'm seeing issues with the movers and with multi-selection in this branch.

Do you think it's possible to use another trick to move the margins to the parent wrapper instead, I'd feel more confident with the changes, if we can do so instead? 

This doesn't mean we can't move these here, just that there are probably consequences I'm not aware of.

--
Also, do you think it's possible to split out some of the enhancements of this PR to separate smaller PRs, this would ease reviewing testing and mergin.

Reply:
Thanks for the thoughts. 

I think you've confirmed for me that it's a good idea to split this up into the general enhancements and refactor, and the collapsing margins separately, to let us focus. I will do that. 

>  Do you think it's possible to use another trick to move the margins to the parent wrapper instead, I'd feel more confident with the changes, if we can do so instead?

I don't know, but I don't think so. In https://codepen.io/joen/pen/aqKLqa I basically tried finding the _minimal_ layout we could use, to keep the block-docked toolbar _and_ allow the margins to collapse. Basically, as soon as the block whose margins need to be allowed to collapse gets an adjecent element, like the block toolbar, those margins uncollapse.